packages/-ember-data/tests/unit/debug-test.js

```diff
@@ -1,105 +1,113 @@
+      'model:post',
+      Model.extend({
+        title: attr('string'),
+      })
+    );
+
+    this.owner.register(
+      'model:user',
+      Model.extend({
+        name: attr('string'),
+        isDrugAddict: attr('boolean'),
+        maritalStatus: belongsTo('marital-status', { async: false }),
+        posts: hasMany('post', { async: false }),
+      })
+    );
+
+    this.owner.register('adapter:application', Adapter.extend({}));
+
+    let record = this.owner.lookup('service:store').createRecord('user');
+
+    let propertyInfo = record._debugInfo().propertyInfo;
+
+    assert.equal(propertyInfo.groups.length, 4);
+    assert.deepEqual(propertyInfo.groups[0].properties, ['id', 'name', 'isDrugAddict']);
+    assert.deepEqual(propertyInfo.groups[1].properties, ['maritalStatus']);
+    assert.deepEqual(propertyInfo.groups[2].properties, ['posts']);
   });
 
-  let record = store.createRecord('user');
-
-  let propertyInfo = record._debugInfo().propertyInfo;
-
-  assert.deepEqual(propertyInfo, {
-    includeOtherProperties: true,
-    groups: [
-      {
-        name: 'Attributes',
-        properties: ['id', 'name', 'isDrugAddict'],
-        expand: true,
-      },
-      {
-        name: 'maritalStatus',
-        properties: ['maritalStatus'],
-        expand: true,
-      },
-      {
-        name: 'posts',
-        properties: ['posts'],
-        expand: true,
-      },
-      {
-        name: 'Flags',
-        properties: ['isLoaded', 'hasDirtyAttributes', 'isSaving', 'isDeleted', 'isError', 'isNew', 'isValid'],
-      },
-    ],
-    expensiveProperties: ['maritalStatus', 'posts'],
+  test('_debugInfo supports arbitray relationship types', function(assert) {
+    this.owner.register(
+      'model:marital-status',
+      Model.extend({
+        name: attr('string'),
+      })
+    );
+
+    this.owner.register(
+      'model:post',
+      Model.extend({
+        title: attr('string'),
+      })
+    );
+
+    this.owner.register(
+      'model:user',
+      Model.extend({
+        name: attr('string'),
+        isDrugAddict: attr('boolean'),
+        maritalStatus: belongsTo('marital-status', { async: false }),
+        posts: computed(() => [1, 2, 3])
+          .readOnly()
+          .meta({
+            options: { inverse: null },
+            isRelationship: true,
+            kind: 'customRelationship',
+            name: 'posts',
+            type: 'post',
+          }),
+      })
+    );
+
+    this.owner.register('adapter:application', Adapter.extend({}));
+
+    let record = this.owner.lookup('service:store').createRecord('user');
+
+    let propertyInfo = record._debugInfo().propertyInfo;
+
+    assert.deepEqual(propertyInfo, {
+      includeOtherProperties: true,
+      groups: [
+        {
+          name: 'Attributes',
+          properties: ['id', 'name', 'isDrugAddict'],
+          expand: true,
+        },
+        {
+          name: 'maritalStatus',
+          properties: ['maritalStatus'],
```

Comment:
How do you feel about keeping things like 
```
const MaritalStatus = DS.Model.extend({
    name: DS.attr('string'),	   
  });	
```
so that searching for the name/defintion of `maritalStatus` will find results

Reply:
yea, I can change those. Initially I thought it was nice to save the vertical space but then I ran prettier and it is actually even longer the other way anyway

Reply:
@igorT i moved the model definitions back outside of the registrations